ledger-service/metrics/src/main/scala/com/digitalasset/http/metrics/HttpJsonApiMetrics.scala

```diff
@@ -40,14 +40,21 @@ class HttpJsonApiMetrics(
 
   @nowarn
   object Db {
-    val prefix: MetricName = HttpJsonApiMetrics.this.prefix :+ "db"
+    private val prefix: MetricName = HttpJsonApiMetrics.this.prefix :+ "db"
 
     val fetchByIdFetch: Timer = defaultMetricsFactory.timer(prefix :+ "fetch_by_id_fetch")
     val fetchByIdQuery: Timer = defaultMetricsFactory.timer(prefix :+ "fetch_by_id_query")
     val fetchByKeyFetch: Timer = defaultMetricsFactory.timer(prefix :+ "fetch_by_key_fetch")
     val fetchByKeyQuery: Timer = defaultMetricsFactory.timer(prefix :+ "fetch_by_key_query")
     val searchFetch: Timer = defaultMetricsFactory.timer(prefix :+ "search_fetch")
     val searchQuery: Timer = defaultMetricsFactory.timer(prefix :+ "search_query")
+    val updateCache: Timer = defaultMetricsFactory.timer(prefix :+ "update_cache")
+    val queryResponseFailureCounter: Counter =
+      defaultMetricsFactory.counter(prefix :+ "failure_response_query_count")
+    val startedUpdateCacheCounter: Counter =
+      defaultMetricsFactory.counter(prefix :+ "started_updating_cache_count")
+    val failUpdateCacheCounter: Counter =
+      defaultMetricsFactory.counter(prefix :+ "failed_updating_cache_count")
```

Comment:
In terms of naming, I think it's good if the component of the system we're measuring is the prefix. That's consistent with the other metrics too. Looking at queries overall, it seems it's measured by the two `Timer` metrics, `searchFetch` and `searchQuery`, which record the latency of the cache refresh and the db query respectively. We can't exactly put those two histograms together to get percentiles of the whole thing, because it doesn't know which fetches go with which queries, but maybe that's okay. We do want counters for the (i) start, (ii) success (or any completion) and (iii) failure of the query overall. We have `queryResponseFailureCounter` for failure, and perhaps we could treat the counter on searchQuery as a counter of successes, as it's roughly the last piece of most queries, but I think it's cleaner to have a dedicated counter for that, and we're still missing a counter on starts, so I'd have something like

```suggestion
        val searchStarted: Counter = defaultMetricsFactory.counter(prefix :+ "search_started")
        val searchFailed: Counter = defaultMetricsFactory.counter(prefix :+ "search_failed")
        val searchCompleted: Counter = defaultMetricsFactory.counter(prefix :+ "search_completed")

        val cacheUpdate: Timer = defaultMetricsFactory.timer(prefix :+ "cache_update") // This includes a counter of completed.
    val cacheUpdateStarted: Counter = defaultMetricsFactory.counter(prefix :+ "cache_update_started")
    val cacheUpdateFailed: Counter = defaultMetricsFactory.counter(prefix :+ "cache_update_failed")
```

Reply:
addressed